mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -786,6 +790,17 @@ bool CMomentumPlayer::CreateStartMark()
     return false;
 }
 
+bool CMomentumPlayer::SetStartMark(int track, SavedLocation_t *saveloc)
```

Comment:
Couple things I noticed about this function is:

1. The result is never used, so this can just be a void.
2. Make sure to early return when doing validation checks, it leads to much cleaner code:

```
if (track < 0 || track >= MAX_TRACKS)
    return;

if (saveloc->m_savedComponents != (SAVELOC_POS | SAVELOC_ANG))
    return;
```

Reply:
Thanks for taking the time to review my commits, wouldn't have spotted these things otherwise. In my latest commit I hope I've fixed all these, but I'll explain why in each requested change before resolving.

1. SetStartMark's bool return was meant to be used for other purposes but there is no reason I shouldn't be using it already, so I've now utilised it to warn the player if something goes wrong.
2. This has now been edited into my functions, it does look much nicer.

Reply:
Feel free to unresolve any of these if you feel I haven't actually fixed them, by the way.

Reply:
The function does not have my suggested change and does not early return. Unresolving it.

Reply:
Not sure how this happened but it's fixed now. I hope.